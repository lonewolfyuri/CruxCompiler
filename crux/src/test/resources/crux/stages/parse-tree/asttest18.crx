// test block-level scope

func main() : void {
  var a : int;
  if (true) {
    var a: bool;
    ::foo();
  }
}
