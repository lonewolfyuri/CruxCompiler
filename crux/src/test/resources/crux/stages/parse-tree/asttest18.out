program
  declarationList
    declaration
      functionDefinition
        parameterList
        type
        statementBlock
          statementList
            statement
              variableDeclaration
                type
            statement
              ifStatement
                expression0
                  expression1
                    expression2
                      expression3
                        expression0
                          expression1
                            expression2
                              expression3
                                literal
                statementBlock
                  statementList
                    statement
                      variableDeclaration
                        type
                    statement
                      callStatement
                        callExpression
                          expressionList
